### PR TITLE
Fix issue with local mode bodiless responses causing clients to hang

### LINKED ISSUE #525: OPTIONS generated by CORS doesn't work correctly in local
First of all, thank you for this awesome framework. Really easy to use, I like it a lot.

I'm developing an API that has to accept CORS and i'm having troubles with the OPTIONS methods.

The problem is easy to reproduce:
In app.py
```
...
@app.route('/test', methods=['POST'], cors=True)
def test_cors():
  return {'hello': 'world'}

```
Then, run it in local: `chalice local --port=8100`

When you try to do the first OPTIONS curl, everything works =)

```
curl -H "Origin: http://example.com" \                                                      
  -H "Access-Control-Request-Method: POST" \
  -H "Access-Control-Request-Headers: X-Requested-With" \
  -X OPTIONS --verbose \
  http://localhost:8100/test
```
Response:
```
*   Trying ::1...
* TCP_NODELAY set
* Connection failed
* connect to ::1 port 8100 failed: Connection refused
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8100 (#0)
> OPTIONS /test HTTP/1.1
> Host: localhost:8100
> User-Agent: curl/7.51.0
> Accept: */*
> Origin: http://example.com
> Access-Control-Request-Method: POST
> Access-Control-Request-Headers: X-Requested-With
>
* HTTP 1.0, assume close after body
< HTTP/1.0 200 OK
< Server: BaseHTTP/0.6 Python/3.6.0
< Date: Thu, 07 Sep 2017 16:02:02 GMT
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Headers: Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key
< Access-Control-Allow-Methods: POST,OPTIONS
<
* Curl_http_done: called premature == 0
* Closing connection 0
```

But the next OPTIONS curl will not end never =(

```
*   Trying ::1...
* TCP_NODELAY set
* Connection failed
* connect to ::1 port 8100 failed: Connection refused
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8100 (#0)
> OPTIONS /test HTTP/1.1
> Host: localhost:8100
> User-Agent: curl/7.51.0
> Accept: */*
> Origin: http://example.com
> Access-Control-Request-Method: POST
> Access-Control-Request-Headers: X-Requested-With
>
< HTTP/1.1 200 OK
< Server: BaseHTTP/0.6 Python/3.6.0
< Date: Thu, 07 Sep 2017 16:02:03 GMT
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Headers: Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key
< Access-Control-Allow-Methods: POST,OPTIONS
* no chunk, no close, no size. Assume close to signal end
```

It stops in  `* no chunk, no close, no size. Assume close to signal end`. It seems like the server only send the close signal only in the first call. If you stop the second curl and try it again, it wont stop neither.
 
Do you have the same problem? Can you reproduce it?

P.S.: I'm using macOS, but I hope that is not a problem to do the local development =P

### PR DESCRIPTION
**Pull request by stealthycoin**
fixes #525

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -2,6 +2,13 @@
 CHANGELOG
 =========
 
+Next Release (TBD)
+==================
+* Fix issue with chalice local mode causing http clients to hang on responses
```
Comment (jamesls) [inline CHANGELOG.rst]: Line between header and changelog entry.

File: tests/integration/test_local.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import socket
```
Comment (jamesls) [inline tests/integration/test_local.py]: Hmm, I'm surprised flake8 didn't complain about this.  Please use pep8 guidelines for grouping imports.

File: chalice/local.py [hunk 3]
```diff
@@ -201,7 +201,7 @@ class NotAuthorizedError(LocalGatewayException):
 
 
 class NoOptionsRouteDefined(LocalGatewayException):
-    CODE = 403
+    CODE = 200
```
Comment (jamesls) [inline chalice/local.py]: Why was this changed?

Comment (stealthycoin) [reply chalice/local.py]: Because thats what API gateway does. I missed it before but I noticed it when going over these the tests for cors cases. I can do this as a separate PR if you would prefer.

Comment (jamesls) [reply chalice/local.py]: I don't see a test for it.  I don't mind it in this PR, but we just have a test for this.  Might also be good to have a separate commit for this with a description of why it's being changed.

Comment (stealthycoin) [reply chalice/local.py]: The existing tests should have covered it and I corrected the tests while I was there anyway and thats when I noticed it was incorrect.

I added these assertions:
* https://github.com/stealthycoin/chalice/blob/52f48152c7d2206c6fe9fe62cc39cf243f52e9fa/tests/unit/test_local.py#L348

* https://github.com/stealthycoin/chalice/blob/52f48152c7d2206c6fe9fe62cc39cf243f52e9fa/tests/unit/test_local.py#L364

Comment (jamesls) [reply chalice/local.py]: I don't follow.  This is updating the code for an exception right?  Those tests are for successful requests aren't they?  Are you saying we were previously returning a 403 response when we should have been returning a 200?

Comment (stealthycoin) [reply chalice/local.py]: The exception is raised when there is no OPTIONS method defined on a route which LocalGateway exceptions are raised when the routes cannot process the request, so that is reasonable so far. The calling code can still generate the OPTIONS request that API Gateway would return and then send a successful response if there is a CORSConfig object on it. I missed that it was a success case somehow before. But looking at it again it seems like it should be decoupled entirely from that type of error since all the others exceptions like this are actual error cases. Whereas this one should be caught and turned into a successful response by auto-generating the CORS headers.

I will fix this in a separate PR.
